src/blockencodings.h

```diff
@@ -40,10 +40,9 @@ class BlockTransactionsRequest {
         uint64_t indexes_size = (uint64_t)indexes.size();
         READWRITE(COMPACTSIZE(indexes_size));
         if (ser_action.ForRead()) {
-            size_t i = 0;
             while (indexes.size() < indexes_size) {
                 indexes.resize(std::min((uint64_t)(1000 + indexes.size()), indexes_size));
-                for (; i < indexes.size(); i++) {
+                for (size_t i = 0; i < indexes.size(); i++) {
```

Comment:
Change this occurrence to `j`? Maybe it would help to combine such changes in a single pull?